nifi-extension-bundles/nifi-github-bundle/nifi-github-extensions/src/main/java/org/apache/nifi/github/GitHubRepositoryClient.java

```diff
@@ -255,20 +258,27 @@ public InputStream getContentFromCommit(final String path, final String commitSh
      * @throws IOException if an I/O error happens calling GitHub
      * @throws FlowRegistryException if a non I/O error happens calling GitHub
      */
-    public List<GHCommit> getCommits(final String path, final String branch) throws IOException, FlowRegistryException {
+    @Override
+    public List<GitCommit> getCommits(final String path, final String branch) throws IOException, FlowRegistryException {
         final String resolvedPath = getResolvedPath(path);
         final String branchRef = BRANCH_REF_PATTERN.formatted(branch);
         LOGGER.debug("Getting commits for [{}] from branch [{}] in repo [{}]", resolvedPath, branch, repository.getName());
 
         return execute(() -> {
             try {
                 final GHRef branchGhRef = repository.getRef(branchRef);
-                return repository.queryCommits()
+                final List<GHCommit> ghCommits = repository.queryCommits()
                         .path(resolvedPath)
                         .from(branchGhRef.getObject().getSha())
                         .pageSize(COMMIT_PAGE_SIZE)
                         .list()
                         .toList();
```

Comment:
As we're iterating / mapping over the result immediately after, we don't need to collect it into a `List` at this point.

```suggestion
                final PagedIterable<GHCommit> ghCommits = repository.queryCommits()
                        .path(resolvedPath)
                        .from(branchGhRef.getObject().getSha())
                        .pageSize(COMMIT_PAGE_SIZE)
                        .list();
```

Reply:
The `toList()` call here is from GH client's `PagedIterable` and its a way to eagerly fetch the whole list, which we want since the NiFi UI currently doesn't have a way to page through the versions of the flows. 

It could be possible to try and return the converted list in one big statement by adding:
```
.stream()
.map(this::toGitCommit)
.toList();
```
But the `toGitCommit` method throws `IOException` because of one of the model classes throwing `IOException`, and then you can't have the checked exception inside the stream statements, so that is why I ended up with the second block of code to convert it to the other list.

Reply:
Yes, I just thought the call to `.toList()` after `.list()` is not necessary as we immediately iterate over all elements to create our own `List` anyway. 
I assumed the iteration could run over the `PagedIterable` directly instead of creating an intermediate `List` object. 

Didn't notice `toGitCommit` throwing an `IOException` making it less suited for the `Stream` api. Thanks for pointing that out.

Reply:
If we don't call `toList()` then `list()` returns one page, and we'd have use the results of the `PagedIterable` to keep making more calls to `repository.queryCommits()` for the rest of the pages, which is exactly what `toList()` is doing for us.

Reply:
I see, thanks for clarifying.